packages/adobe-sign/__tests__/repos/base.repo.spec.ts

```diff
@@ -0,0 +1,120 @@
+import "reflect-metadata";
+import { DynamoProvider, EnvironmentVariable, Mock, NotFoundError } from "asu-core";
+import { TemplatesRepo } from "../../src/repos/templates.repo";
+import { DynamoDB, Service, Request } from "aws-sdk";
+import { PutItemOutput, GetItemOutput } from "aws-sdk/lib/dynamodb/types";
+import { AWSError } from "aws-sdk/lib/error";
+import { Template } from "../../src/models/asu/template";
+
+describe('TemplatesRepo', () => {
+  const expectedStage = "dev";
+
+  function setup() {
+    const connectionProvider = Mock(new DynamoProvider());
+    const dynamo = Mock(new DynamoDB.DocumentClient());
+    const repo = new TemplatesRepo(connectionProvider);
+
+    connectionProvider.resolve.mockReturnValue(dynamo);
+
+    return { repo, dynamo }
+  }
+
+  beforeEach(() => {
+    process.env = {};
+    process.env[EnvironmentVariable.Stage] = expectedStage;
+  });
+
+  const putReturns = (dynamo: jest.Mocked<DynamoDB.DocumentClient>, returnValue?: any) => {
+    const putReturn = Mock(new Request<PutItemOutput, AWSError>(Mock(new Service()), null));
+    putReturn.promise.mockResolvedValue(returnValue || null);
+    dynamo.put.mockReturnValue(putReturn);
+  };
+
+  const getReturns = (dynamo: jest.Mocked<DynamoDB.DocumentClient>, returnValue: any) => {
+    const getReturn = Mock(new Request<GetItemOutput, AWSError>(Mock(new Service()), null));
+    getReturn.promise.mockResolvedValue(returnValue || null);
+    dynamo.get.mockReturnValue(getReturn);
+  };
+
+  /**
+   * BaseRepo is an abstract class so we are testing it through
+   * the TemplatesRepo.
+   */
+  describe('create', () => {
+    it('should throw when dynamo throws', async () => {
+      // Arrange
+      const { repo, dynamo } = setup();
+      const expectedError = new Error("Error");
+      const template = new Template();
+
+      dynamo.put.mockImplementation(() => { throw expectedError; });
+
+      // Act / Assert
+      expect(repo.create(template)).rejects.toBe(expectedError);
```

Comment:
missing `await` for the async `create` call and might I also suggest using `toThrow` here since we're testing an Error?
```suggestion
      await expect(repo.create(template)).rejects.toThrow(expectedError);
```